editor/connections_dialog.cpp

```diff
@@ -1446,6 +1446,9 @@ void ConnectionsDock::update_tree() {
 			if (!doc_class_name.is_empty() && !doc_data->class_list.find(doc_class_name)) {
 				doc_class_name = String();
 			}
+			if (doc_class_name.is_empty()) {
+				doc_class_name = script_base->get_path().get_file().get_basename();
+			}
```

Comment:
The `gds` here is not empty, so it won't enter this condition. Based on current testing, there haven't been any issues caused by this.
Maybe adding a check for the script type would be a good idea?

Reply:
If it's not empty for GDScript, then it should probably not be empty for C# scripts either so maybe it's a bug in the C# scripting implementation. But it seems unrelated to the tooltip feature anyway.

Reply:
This should be fixed with the changes described in https://github.com/godotengine/godot/pull/83505#discussion_r1774261467. Using the correct doc class name in the generator, `doc_class_name` won't be empty for C# (same as GDScript).

Reply:
Even if something had to be changed in this file, it should have been added above lines 1446-1448, since this checks that a class with that `doc_class_name` exists in `doc_data`. If the class does not exist, then `doc_class_name` is cleared so that the "Open Documentation" item in the context menu is disabled.

But I don't think there is a mistake. The logic is as follows:

1. If the script has a global name, then its `doc_class_name` is equal to the global name.
2. Otherwise, the `doc_class_name` is equal to script path, without the `res://` prefix, in quotes.[^1]

Accordingly, you need to reproduce the same logic in C# when generating documentation.

[^1]: Inner classes are not taken into account since they cannot be attached to a node/resource via the editor, only via code. This may not be true for tool scripts, but that's a separate issue.